docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,124 @@
+	// +optional
+	IPBlocks []*IPBlock `json:"ipBlocks,omitempty"`
+}
+```
+The mapping the `Selector` elements into the `from` NP ingress definitions will be done according the following algorithm:
+- if `Namespaces` is empty, a [NetworkPolicyPeer](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#networkpolicypeer-v1-networking-k8s-io)
+  with `podSelector` equals to `WorkloadSelector` will be created. Which means that only pods with the defined labels from 
+  the module namespace will be able to connect, or be connected.
+- If `Namespaces` is not empty, for each entry in the array will be created a `NetworkPolicyPeer`
+   with `podSelector` equals to `WorkloadSelector` and a `namespaceSelector` contains a single `kubernetes.io/metadata.name`
+   label equals to the entry from `Namespaces`. [Here](https://kubernetes.io/docs/concepts/services-networking/network-policies/#targeting-a-namespace-by-its-name) 
+   you can read more about targeting a Namespace by its name. If `Namespaces` is not empty but `WorkloadSelector` is empty, 
+   then any pod from the selected namespaces will be able to connect.
+
+- If `IPBlocks` is not empty, a separate NetworkPolicyPeer will be created for each IPBlocks entry.
+
+### FybrikModule extensions
+
+Some of Fybrik modules need external services for their functionality. For example, a currency exchange module checks 
+the currency rates form an external service.
+We suggest adding an `externalServices` array into the `FybrikModule.Spec`. Each entry will be a DNS name or IP address of 
+the external service. During `egress` NetworkPolicyPeer creations, each service name will be translated to a single IPBlock entry.
+
+### PlotterController extensions
+
+In order to create ingress policies for modules, the blueprint that creates NPs should know potential clients, therefore
+PlotterController who has the entire picture, should provide this information. 
+
+NetworkPolicies select pods according to labels. In order to separate policies for 2 different modules, each module should
+have a unique label, in additional to the common, fybrik application defined labels.
+
+### BlueprintController extensions
+
+BlueprintController installs and uninstalls Fybrik modules. We suggest extending this functionality by creating an instance
+of NP for each installing module, and delete it when the module is uninstalled.
+
+For the data plane entry module, the ingress element will be created based on information from a 
+relevant `FybrikApplication`. For other modules this information will be provided by the `PlotterController`
+Fybrik will not support different network restrictions to different module ports. (NP allow it)
+
+The egress element will be a combination of destinations, such as, the next module (`2a`), data source (`5`), Fybrik (`3`) 
+or module required (`4`) services. The connection types are taken from the [network topologies](FybrikNetworkTopologiesAndRequirements.md)
```

Comment:
How are these destinations formed? Specifically:
2a - Fybrik knows where the module is deployed (namespace) but not the labels of the deployed pods which is internal to the module chart implementation. Unless I am missing some assumptions on module implementation.
3 - any change to the existing deployment of Vault and the relevant policies?
5 - Fybrik is not aware of the data store destination. The connection details are passed as-is to the access module. How do you see this policy implemented?

Reply:
2a, it is a regular chaining convection, fybrik adds some of the labels, and actually for NP I need only a single one `app.kubernetes.io/instance`, I can demonstrate how it works
3- I don't see any changes
5 - I plan to get the destination details from the connection details